conda_gitenv/tests/integration/test_deploy.py

```diff
@@ -48,20 +55,24 @@ def test(self):
         with resolve.tempdir() as tmpdir:
             deploy.deploy_repo(self.repo, tmpdir)
 
-            def link_target(env, label):
-                link = os.path.join(tmpdir, env, label)
-                target = os.path.join(tmpdir, env, os.readlink(link))
-                return target 
-
-            self.assertTrue(os.path.exists(link_target('default', 'next')))
-            self.assertTrue(os.path.exists(link_target('default', 'current')))
-            self.assertTrue(os.path.exists(link_target('bleeding', 'next')))
-            self.assertTrue(os.path.exists(link_target('bleeding', 'latest')))
-            self.assertTrue(os.path.exists(link_target('default', 'latest')))
+            self.assertTrue(os.path.exists(link_target(tmpdir, 'default', 'next')))
+            self.assertTrue(os.path.exists(link_target(tmpdir, 'default', 'current')))
+            self.assertTrue(os.path.exists(link_target(tmpdir, 'bleeding', 'next')))
+            self.assertTrue(os.path.exists(link_target(tmpdir, 'bleeding', 'latest')))
+            self.assertTrue(os.path.exists(link_target(tmpdir, 'default', 'latest')))
 
             # Check that we can resolve those links, finding the python executable.
             self.assertTrue(os.path.exists(os.path.join(tmpdir, 'bleeding', 'next', 'bin', 'python')))
 
+    def test_specified_env_labels(self):
+        with resolve.tempdir() as tmpdir:
+            deploy.deploy_repo(self.repo, tmpdir, ['default/next', 'bleeding/latest'])
+
+            self.assertTrue(os.path.exists(link_target(tmpdir, 'default', 'next')))
+            self.assertTrue(os.path.exists(link_target(tmpdir, 'bleeding', 'latest')))
```

Comment:
Could you make an assertion that some environments *don't* exist?

(edit: The ```os.readlink``` does make things awkward, and in reality you have already done this negative assertion in the catching of the error)